extensions/interactions/NumericInput/NumericInput.js

```diff
@@ -79,3 +94,38 @@ oppia.directive('oppiaShortResponseNumericInput', [
     };
   }
 ]);
+
+oppia.factory('numericInputService', [
+    'interactionService', function(interactionService) {
+  var RULES = {
+    'Equals': function(answer, inputs) {
+      return answer == inputs.x;
+    },
+    'IsLessThan': function(answer, inputs) {
+      return answer < inputs.x;
+    },
+    'IsGreaterThan': function(answer, inputs) {
+      return answer > inputs.x;
+    },
+    'IsLessThanOrEqualTo': function(answer, inputs) {
+      return answer <= inputs.x;
+    },
+    'IsGreaterThanOrEqualTo': function(answer, inputs) {
+      return answer >= inputs.x;
+    },
+    'IsInclusivelyBetween': function(answer, inputs) {
+      return answer >= inputs.a && answer <= inputs.b;
+    },
+    'IsWithinTolerance': function(answer, inputs) {
+      return answer >= inputs.x - inputs.tol &&
+      answer <= inputs.x - inputs.tol;
```

Comment:
Nit (style): Multi-line expressions should be wrapped in parentheses. Subsequent lines after the first should be indented, e.g.:

```
      return (answer >= inputs.x - inputs.tol &&
        answer <= inputs.x - inputs.tol);
```